### PR TITLE
Remove search config selector hacks in dim-api reducer

### PR DESCRIPTION
**Pull request by robojumper**
nev and I found this pretty ugly selector hack that's not really needed.

This conflicts with #9125 so I will rebase after that one merges, I just didnt't want to forget about this.

### CODE REVIEW

**Review by bhollis — APPROVED** [review]
This is great, thanks!

File: src/app/dim-api/reducer.ts [hunk 1]
```diff
@@ -1174,10 +1161,13 @@ function saveSearch(
...
   const destinyVersion = account.destinyVersion;
-  const searchConfigs = searchConfigSelector(stubSearchRootState(account));
+  // Note: memoized
+  const filtersMap = buildFiltersMap(destinyVersion);
 
   // Canonicalize the query so we always save it the same way
-  const { canonical, saveable } = parseAndValidateQuery(query, searchConfigs);
+  const { canonical, saveable } = parseAndValidateQuery(query, filtersMap, {
+    customStats: draft.settings.customStats ?? [],
+  } as FilterContext);
```
Comment (robojumper) [inline src/app/dim-api/reducer.ts:1170]: @nev-r was this (and the one below) simply forgotten about?

Comment (nev-r) [reply src/app/dim-api/reducer.ts:1170]: not sure how you mean. pretty sure this is what it took to get it to believe custom string named stats were valid filters in all cases

Comment (robojumper) [reply src/app/dim-api/reducer.ts:1170]: Apologies if it's unsure from the diff but right now that code doesn't pass any custom stat names here and I can't seem to save queries with custom stat names on beta.

Comment (nev-r) [reply src/app/dim-api/reducer.ts:1170]: i see completely what you mean, thank you. same should have happened below.
it looks kinda control-z'd at some point, merging with master was really hectic.

Comment (nev-r) [reply src/app/dim-api/reducer.ts:1170]: should be good in master

Comment (nev-r) [reply src/app/dim-api/reducer.ts:1170]: actually maybe you mean something else with the original comment... saving custom stats is now working in master though

**Comment by bhollis** [comment]
Anything left on this?

**Comment by robojumper** [comment]
Some expected conflicts with other open PRs. But we can also merge this first if you'd like

**Pull request merged**